File: ServerApp/Entities/Periods/Semester/SemesterCourseLevelSpeciality.cs
using Everest.AspNetStartup.Binding;
using Everest.AspNetStartup.Models;
using Exam.Entities.Courses;
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;

namespace Exam.Entities.Periods
{
    [ModelBinder(BinderType = typeof(ItemValueModelBinder))]
    public class SemesterCourseLevelSpeciality:Entity<long>
    {
        [JsonIgnore]
        public virtual SemesterCourse SemesterCourse { get; set; }
        public long? SemesterCourseId { get; set; }


        [JsonIgnore]
        public virtual SemesterLevelSpeciality SemesterLevelSpeciality { get; set; }
        public long SemesterLevelSpecialityId { get; set; }

        [JsonIgnore]
        public virtual CourseLevelSpeciality CourseLevelSpeciality { get; set; }
        public long? CourseLevelSpecialityId { get; set; }
    }
}

File: webapp/apps/regolia/src/models/entities/course/semester-course-level-speciality.ts
import {Entity} from "../entity";
import {SemesterLevelSpeciality} from "../semester";
import {SemesterCourse} from "./semester-course";
import {CourseLevelSpeciality} from "./course-level-speciality.entity";

export class SemesterCourseLevelSpeciality extends Entity<number> {
  constructor(value: any = {}) {
    super();

    this.id = value.id;
    this.registrationDate = new Date(value.registrationDate);

    this.semesterCourseId = value.semesterCourseId;
    this.courseLevelSpecialityId = value.courseLevelSpecialityId;
    this.semesterLevelSpecialityId = value.semesterLevelSpecialityId;

    if(value.semesterCourse) {
      this.semesterCourse = new SemesterCourse(value.semesterCourse);
    }
  }

  courseLevelSpeciality: CourseLevelSpeciality;
  courseLevelSpecialityId: number;

  semesterCourse: SemesterCourse;
  semesterCourseId: number;

  semesterLevelSpeciality: SemesterLevelSpeciality;
  semesterLevelSpecialityId: number;
}
